tests/integration/session/test_sign_out_and_exit.py

```diff
@@ -0,0 +1,157 @@
+from app.helpers.template_helpers import (
+    CENSUS_CY_BASE_URL,
+    CENSUS_EN_BASE_URL,
+    CENSUS_NIR_BASE_URL,
+)
+from tests.integration.integration_test_case import IntegrationTestCase
+
+SIGN_OUT_URL_PATH = "/sign-out"
+SIGNED_OUT_URL_PATH = "/signed-out"
+ACCOUNT_SERVICE_LOG_OUT_URL = "http://localhost/logout"
+LOG_OUT_URL_PATH = "/logout"
+
+
+class TestSaveAndSignOut(IntegrationTestCase):
+    # Test the behaviour when using a Hub/No Hub
+
+    def test_no_session_cookie_redirects_to_census_home_page(self):
+        for schema in ["test_textfield", "test_hub_and_spoke"]:
+            with self.subTest(schema=schema):
+                self.get(SIGN_OUT_URL_PATH, follow_redirects=False)
+                self.assertInRedirect(CENSUS_EN_BASE_URL)
+
+    def test_no_account_service_log_out_url_redirects_to_signed_out_page(self):
+        for schema in ["test_textfield", "test_hub_and_spoke"]:
+            with self.subTest(schema=schema):
+                self.launchSurvey(schema)
+                self.get(SIGN_OUT_URL_PATH)
+                self.assertInUrl(SIGNED_OUT_URL_PATH)
+
+    def test_redirects_to_account_service_log_out_url_when_present(self):
+        for schema in ["test_textfield", "test_hub_and_spoke"]:
+            with self.subTest(schema=schema):
+                self.launchSurvey(
+                    schema, account_service_log_out_url=ACCOUNT_SERVICE_LOG_OUT_URL
+                )
+                self.get(SIGN_OUT_URL_PATH)
+                self.assertInUrl(LOG_OUT_URL_PATH)
+
+
+class TestExitPostSubmissionTestCase(IntegrationTestCase):
+    def _launch_and_submit_questionnaire(self, schema, **kwargs):
+        self.launchSurvey(schema, **kwargs)
+        self.post()
+        self.post()
+        self.assertInUrl("/thank-you")
+
+
+class TestExitPostSubmissionWithFinalSummaryDefaultTheme(
+    TestExitPostSubmissionTestCase
+):
+    def test_no_account_service_log_out_url_redirects_to_signed_out_page(self):
+        self._launch_and_submit_questionnaire(schema="test_textfield")
+        self.get(SIGN_OUT_URL_PATH)
+        self.assertInUrl(SIGNED_OUT_URL_PATH)
+
+    def test_redirects_to_account_service_log_out_url_when_present(self):
+        self._launch_and_submit_questionnaire(
+            schema="test_textfield",
+            account_service_log_out_url=ACCOUNT_SERVICE_LOG_OUT_URL,
+        )
+        self.get(SIGN_OUT_URL_PATH)
+        self.assertInUrl(LOG_OUT_URL_PATH)
```

Comment:
It would be clearer if this was named `ACCOUNT_SERVICE_LOG_OUT_URL_PATH`

Reply:
Done